apps/io-services-cms-webapp/src/reviewer/__tests__/request-review-handler.test.ts

```diff
@@ -0,0 +1,329 @@
+  total: 12,
+  issues: [aJiraIssue],
+};
+
+const anEmptySearchJiraIssuesResponse: SearchJiraIssuesResponse = {
+  ...aSearchJiraIssuesResponse,
+  issues: [],
+};
+
+const aCreateJiraIssueResponse = {
+  id: aJiraIssue.id,
+  key: aJiraIssue.key,
+} as unknown as CreateJiraIssueResponse;
+
+const aService = {
+  id: "aServiceId",
+  data: {
+    name: "aServiceName" as NonEmptyString,
+    description: "aServiceDescription",
+    authorized_recipients: [],
+    max_allowed_payment_amount: 123,
+    metadata: {
+      address: "via tal dei tali 123",
+      email: "service@email.it",
+      pec: "service@pec.it",
+      scope: "LOCAL",
+    },
+    organization: {
+      name: "anOrganizationName",
+      fiscal_code: "12345678901",
+    },
+    require_secure_channel: false,
+  },
+} as unknown as Service;
+
+const aDelegate = {
+  firstName: "aName",
+  lastName: "aSurname",
+  email: "anEmail",
+  permissions: ["permission1", "permission2"],
+} as Delegate;
+
+const anInsertQueryResult: QueryResult = {
+  command: "string",
+  rowCount: 1,
+  oid: 1,
+  fields: [],
+  rows: [],
+};
+
+const aDbInsertData = {
+  service_id: aService.id,
+  service_version: aService.id,
+  ticket_id: aCreateJiraIssueResponse.key,
+  status: "PENDING",
+  extra_data: "d" as NonEmptyString,
+};
+
+const aVoidFn = () => console.log("");
+
+const mainMockServiceReviewDao = {
+  insert: vi.fn((data: ServiceReviewRowDataTable) => {
+    return TE.of(anInsertQueryResult);
+  }),
+  executeOnPending: vi.fn(
+    (
+      fn: (items: ServiceReviewRowDataTable[]) => TE.TaskEither<Error, void>
+    ) => {
+      return TE.of(aVoidFn());
+    }
+  ),
+};
+
+const mainMockJiraProxy = {
+  createJiraIssue: vi.fn((service: Service, delegate: Delegate) => {
+    return TE.of(aCreateJiraIssueResponse);
+  }),
+  searchJiraIssuesByKey: vi.fn(
+    (jiraIssueKeys: ReadonlyArray<NonEmptyString>) => {
+      return TE.of(aSearchJiraIssuesResponse);
+    }
+  ),
+  getJiraIssueByServiceId: vi.fn((serviceId: NonEmptyString) => {
+    return TE.of(O.none);
+  }),
+};
+
+const mainMockApimProxy = {
+  getDelegateFromServiceId: vi.fn((sid: NonEmptyString) => {
+    return TE.of(aDelegate);
+  }),
+};
+
+describe("Service Review Handler", () => {
+  it("should create a non existing ticket for a new service review", async () => {
+    const handler = createRequestReviewHandler(
+      mainMockServiceReviewDao,
+      mainMockJiraProxy,
+      mainMockApimProxy
+    );
+
+    const context = createContext();
+    const result = await handler(context, JSON.stringify(aService));
+
+    expect(mainMockJiraProxy.getJiraIssueByServiceId).toBeCalledWith(
+      aService.id
+    );
+    expect(mainMockApimProxy.getDelegateFromServiceId).toBeCalledWith(
+      aService.id
+    );
+    expect(mainMockJiraProxy.createJiraIssue).toBeCalledWith(
+      aService,
+      aDelegate
+    );
+    expect(mainMockServiceReviewDao.insert).toBeCalledWith(aDbInsertData);
+    expect(E.isRight(E.of(result)));
```

Comment:
why are we doing this? This sentence is always true, because an E.of(anything) is always E.right....am I wrong?